salt/netapi/__init__.py

```diff
@@ -157,6 +157,11 @@ def run(self, low):
                 "Invalid client specified: '{}'".format(low.get("client"))
             )
 
+        if low.get("client") not in self.opts.get("netapi_enable_clients"):
```

Comment:
I really like this approach, but I'm still of the opinion that we should add a deprecation notice for salt-api users since this will require users to update their configs. 

ping @dwoz thoughts?

Reply:
Where/how would you add a deprecation notice?

Reply:
I'm on the fence here. I don't like subverting out deprecation policy. However, Going through a deprecation path would require us to support both ways for some time. I'd *think* I'd prefer we highlight this change in the release notes. Can we make the case that the security benefits warrant making the change?

Reply:
I definitely agree the feature is very important from a security perspective. Let me bring this change to the entire team to discuss next week and I'll update here

Reply:
The team is in agreement with merging this in, but we want to make sure we document this very thoroughly and improve some of the error messaging. I'll add some comments exactly where we want to add that information